drivers/serial/uart_sam.c

```diff
@@ -12,53 +14,51 @@
  * - The driver works only in polling mode, interrupt mode is not implemented.
  */
 
+#define DT_DRV_COMPAT atmel_sam_uart
+
 #include <errno.h>
 #include <sys/__assert.h>
 #include <device.h>
 #include <init.h>
 #include <soc.h>
 #include <drivers/uart.h>
 
-/*
- * Verify Kconfig configuration
- */
-
-#if CONFIG_UART_SAM_PORT_0 == 1
+#if DT_HAS_NODE(DT_NODELABEL(uart0))
 
-#if DT_UART_SAM_PORT_0_BAUD_RATE == 0
-#error "DT_UART_SAM_PORT_0_BAUD_RATE has to be bigger than 0"
+#if DT_PROP(DT_NODELABEL(uart0), current_speed) == 0
+#error "uart0 current-speed property has to be bigger than 0"
 #endif
```

Comment:
Maybe these can be moved into the block that does `UART_SAM_DEVICE_INIT(n);` or even better into `UART_SAM_DEVICE_INIT` macro itself using `BUILD_ASSERT`?

Reply:
Done! it was need create a macro to check if prop exists. This way we can avoid long list of errors when property not exists.